pkgs/unified_analytics/lib/src/log_handler.dart

```diff
@@ -275,15 +275,20 @@ class LogHandler {
     var records = logFile.readAsLinesSync();
     final content = '${jsonEncode(data)}\n';
 
-    // When the record count is less than the max, add as normal;
-    // else drop the oldest records until equal to max
-    if (records.length < kLogFileLength) {
-      logFile.writeAsStringSync(content, mode: FileMode.writeOnlyAppend);
-    } else {
-      records.add(content);
-      records = records.skip(records.length - kLogFileLength).toList();
-
-      logFile.writeAsStringSync(records.join('\n'));
+    try {
+      // When the record count is less than the max, add as normal;
+      // else drop the oldest records until equal to max
+      if (records.length < kLogFileLength) {
+        logFile.writeAsStringSync(content, mode: FileMode.writeOnlyAppend);
+      } else {
+        records.add(content);
+        records = records.skip(records.length - kLogFileLength).toList();
+
+        logFile.writeAsStringSync(records.join('\n'));
+      }
+    } on FileSystemException {
+      // Logging isn't important enough to warrant raising a
+      // FileSystemException that will surprise consumers of this package.
```

Comment:
I considered adding logic that would retry once after purging some number of records, but I'm not sure if the complexity of this would be worth it.

Reply:
Yeah, let's drop it. The log file is way less important than:

1. ensuring the client tools don't crash
2. sending analytics